lib/error/invalid-href.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+/**
+ * Produces an Erorr object with a message that indicates which resource type has
+ * violated an href validation.
+ *
+ * @param      {string}  href          The href that was passed into a resource getter.
+ * @param      {string}  resourceName  The human readable name of the expected resource type.
+ * @return     {Error}   A simple error with the message
+ */
+function InvalidHrefError(href, resourceName){
+  var message = 'Argument \'href\' (' + href + ') is not a valid ' + resourceName +' href.';
+  return new Error(message);
```

Comment:
@robertjd I really like the approach of having an `InvalidHrefError`! But a question. Any reason why you return a new `Error` from a constructor rather than have the `InvalidHrefError` extend from Error? I find returning things from a constructor to be bad practice and a little confusing. Mostly since the type you'd expect isn't what you'd get, i.e. `err instanceof InvalidHrefError` would fail. And you allocate two objects instead of just one.